app/controllers/PaymentsController.scala

```diff
@@ -49,5 +55,29 @@ class PaymentsController @Inject()(
     }
   }
 
+  private def getOptLastReturn(utr: String, internalId: String)
+                                     (implicit headerCarrier: HeaderCarrier) = {
+    val lastReturnPeriod = ReturnPeriod(LocalDate.now).previous
+    val getOptLastReturn = sdilConnector.returns_get(utr, lastReturnPeriod, internalId)
+    getOptLastReturn
+  }
+
+    private def getOptLastReturnAmount(sdilRef: String, internalId: String)
```

Comment:
I believe there is a neater way to do this without using try/catch and if/else. I will take a look at this